### PR TITLE
Do not redirect if the user does not have view permissions

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @andrewconner @camhashemi 
this side-steps the issue where Eliza does not respect permissions by refusing to open the extension for people that do not have view permissions

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala [hunk 1]
```diff
@@ -112,19 +112,22 @@ class SlackAuthRouter @Inject() (
...
+              Ok(html.maybeExtDeeplink(DeepLinkRedirect(url = keep.url, externalLocator = Some(s"/messages/${pubId.id}")), noExtUrl = keep.url))
+            case _ if !isLoggedIn && onKifi =>
+              val orgIdOpt = keep.organizationId.orElse(slackTeamRepo.getBySlackTeamId(slackTeamId).flatMap(_.organizationId))
+              val orgOpt = orgIdOpt.map(orgId => orgRepo.get(orgId))
+              orgOpt
+                .map(org => redirectThroughSlackAuth(org, slackTeamId, keepPageUrl, keepId = Some(pubId)))
+                .getOrElse(Redirect(keepPageUrl))
+            case _ =>
+              Redirect(keep.url)
+          }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:130]: @camhashemi this is your show, I wasn't sure if the `match` is easier to read than the `if-else` chain. Thoughts?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:130]: I'd `if`/`else` here. matching on Unit is quite inventive :)

**Comment by andrewconner** [comment]
Could we also add a sanity check when fetching a thread on Eliza?

**Comment by camhashemi** [comment]
@andrewconner that's the best way to cover all forbidden paths to the ext but it would require some ext work to handle when it can't fetch the thread. Ideally without any work, not returning a thread would just take you to the third party url and the ext won't open; worst case the ext will open with a blank pane or error state.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala [hunk 2]
```diff
@@ -112,19 +112,22 @@ class SlackAuthRouter @Inject() (
       Keep.decodePublicId(pubId)
         .map(keepId => keepRepo.get(keepId)).filter(_.isActive)
         .map { keep =>
-          val isLoggedIn = request.userIdOpt.isDefined
+          val isLoggedIn = request.userIdOpt.isDefined && permissionCommander.getKeepPermissions(keep.id.get, request.userIdOpt).contains(KeepPermission.VIEW_KEEP)
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:115]: I would separate these two for semantic clarity, `isLoggedIn & hasPermission` or at least rename the variable

**Comment by andrewconner** [comment]
Agreed, but at least it wouldn't show the thread. To be clear, "the" permission check shouldn't be when fetching the thread — it's a failsafe (and lets us know if we have leaks elsewhere)

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:130]: I agree that `if/else` is my dad's way of conditional branching, but it's more straightforward. When pattern matching extracts some information I favor it.

**Comment by camhashemi** [comment]
gotcha, yeah I can add that

**Comment by camhashemi** [comment]
looks good, but I think the `PermissionCommander.getKeepsPermissions` method needs to be updated to see if you have a `ktu` on the keep, can we add that here?

**Comment by ryanpbrewster** [comment]
It already does that.

**Comment by camhashemi** [comment]
confirmed, :+1:

**Pull request closed without merging**